presto-spi/src/test/java/io/prestosql/spi/block/BaseBlockEncodingTest.java

```diff
@@ -15,19 +15,87 @@
 
 import io.airlift.slice.DynamicSliceOutput;
 import io.prestosql.spi.type.Type;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
+import java.util.Arrays;
+import java.util.Random;
+import java.util.stream.IntStream;
+
 import static io.prestosql.spi.block.BlockTestUtils.assertBlockEquals;
 
 @Test(singleThreaded = true)
```

Comment:
these annotations don't work in abstract classes in `TestNG`: https://github.com/cbeust/testng/issues/144

Reply:
And that is why I prefer JUnit - separate instance for every test.
Added a dedicated random in every test. Better than 7 `@Test(singleThreaded = true)` annotations.

Reply:
remove this annotation here then